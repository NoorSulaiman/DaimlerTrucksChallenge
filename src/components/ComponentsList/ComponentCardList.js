import React, { useState, useEffect } from "react";
import styles from "./ComponentCard.module.css";
import {
  Card,
  CardActions,
  CardContent,
  Typography,
  Box,
  Grid,
  Button,
  Dialog,
  DialogActions,
  DialogTitle,
} from "@mui/material";

export default function ComponentCardList({ components, cardButtonFunction }) {
  const [open, setOpen] = useState(false);
  const [locationName, setLocationName] = useState("");
  const [componentsArray, setComponentsArray] = useState(components);
  const onDeleteClick = (name) => {
    setLocationName(name);
    setOpen(true);
  };
  const onDialogYesClick = () => {
    cardButtonFunction(locationName);
    let newComponentsArray = componentsArray;
    componentsArray.map((component, indx) => {
      if (locationName === component.location) {
        newComponentsArray.splice(indx, 1);
        setComponentsArray(newComponentsArray);
      }
    });
    setOpen(false);
  };
  const handleDialogClose = () => {
    setOpen(false);
  };
  useEffect(() => {
    setComponentsArray(components);
  }, [components]);
  const componentCards = componentsArray.map((component) => {
    return (
      <Grid key={component.location} item xs={1}>
        <Card>
          <CardContent>
            <Typography className={styles.itemName}>
              {component.location}
            </Typography>
          </CardContent>
          <CardActions>
            <Button
              onClick={() => onDeleteClick(component.location)}
              className={styles.cardDeleteButton}>
              Delete
            </Button>
          </CardActions>
        </Card>
      </Grid>
    );
  });
  return (
    <div>
      <Box sx={{ flexGrow: 1 }}>
        <Grid container spacing={6}>
          {componentCards}
        </Grid>
      </Box>
      <Dialog
        open={open}
        onClose={handleDialogClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description">
        <DialogTitle id="alert-dialog-title">
          {`Are you sure you want to delete ${locationName} ?`}
        </DialogTitle>
        <DialogActions>
          <Button onClick={handleDialogClose}>No</Button>
          <Button onClick={onDialogYesClick} autoFocus>
            Yes
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
