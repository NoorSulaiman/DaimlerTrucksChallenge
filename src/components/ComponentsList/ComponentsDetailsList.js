import React from "react";
import {
  Accordion,
  AccordionSummary,
  AccordionDetails,
  Typography,
  Box,
  Grid,
} from "@mui/material";
import ExpandMoreIcon from "@mui/icons-material/ExpandMore";
import DetailsTable from "../DetailsTable/DetailsTable";

export default function ComponentsDetailsList({ components }) {
  const componentCards = components.map((component) => {
    return (
      <Grid item xs={5}>
        <Accordion>
          <AccordionSummary
            expandIcon={<ExpandMoreIcon />}
            aria-controls="panel1a-content"
            id="panel1a-header">
            <Typography>{component.location}</Typography>
          </AccordionSummary>
          <AccordionDetails>
            <DetailsTable component={component} />
          </AccordionDetails>
        </Accordion>
      </Grid>
    );
  });
  return (
    <div>
      <Box sx={{ flexGrow: 1 }}>
        <Grid container spacing={6}>
          {componentCards}
        </Grid>
      </Box>
    </div>
  );
}
