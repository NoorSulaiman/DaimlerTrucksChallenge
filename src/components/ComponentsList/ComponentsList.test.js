import React from "react";
import { shallow } from "enzyme";
import ComponentCardList from "./ComponentCardList";
import ComponentsDetailsList from "./ComponentsDetailsList";
const compArray = [
  {
    panelNumber: "22-58883-000",
    panelDescription: "PNL-INR TRIMPLTKAHLUA SPRKL",
    location: "AGAAA",
    parentPanelNumber: "734-C02202",
    modelCode: "H",
    whitelist: ["AGAAA"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "",
    bom: "734-C02202",
    isAsBuilt: true,
  },
  {
    panelNumber: "22-69410-000",
    panelDescription: "PNL-SW,SMARTPLX,OVHD,SD,KAHLUA",
    location: "OVLHP",
    parentPanelNumber: "679-C01232",
    modelCode: "H",
    whitelist: ["OVLHP", "OVRHP"],
    blacklist: [],
    type: "hardwired-component",
    switchId: "",
    bom: "679-C01232",
    isAsBuilt: true,
  },
];

describe("ComponentCardList", () => {
  it("should render ComponentCardList correctly with given props", () => {
    const component = shallow(<ComponentCardList components={compArray} />);
    expect(component).toMatchSnapshot();
  });

  it("should render components cards", () => {
    const component = shallow(<ComponentCardList components={compArray} />);
    expect(component.find("Grid").children.length).toBe(1);
  });
});
describe("ComponentsDetailsList", () => {
  it("should render ComponentCardList correctly with given props", () => {
    const component = shallow(<ComponentsDetailsList components={compArray} />);
    expect(component).toMatchSnapshot();
  });

  it("should render components cards", () => {
    const component = shallow(<ComponentsDetailsList components={compArray} />);
    expect(component.find("Grid").children.length).toBe(1);
  });
});
