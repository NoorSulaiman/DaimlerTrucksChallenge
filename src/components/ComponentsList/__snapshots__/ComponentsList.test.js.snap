// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentCardList should render ComponentCardList correctly with given props 1`] = `ShallowWrapper {}`;

exports[`ComponentsDetailsList should render ComponentCardList correctly with given props 1`] = `ShallowWrapper {}`;
